### ISSUE TITLE
Detect if mbstring is available and output an explicit error message if it isn't

### ISSUE THREAD

**Issue by olivierbellone** [stripe/stripe-php#251]
A relatively common issue with the PHP bindings is that they depend on the `mbstring` extension ([here](https://github.com/stripe/stripe-php/blob/e40211d1d5325d99d3ce167c9ee1ade0473a608a/lib/Util/Util.php#L119)), which isn't necessarily available everywhere.

In particular, on RedHat / CentOS systems, it doesn't come out of the box with PHP and needs to be installed via a separate package.

I was thinking it could be helpful to add a check when the bindings are loaded. Something like:
```php
if (!extension_loaded('mbstring')) {
    die("It looks like the mbstring extension is not loaded. "
      . "The Stripe PHP bindings cannot work properly without this extension. "
      . "Please ask your system administrator to enable the mbstring extension, "
      . "or email support@stripe.com if you have any questions.");
}
```

called from `init.php`.

While we're at it, we could do a similar check for the `curl` extension (although it's much less common for this extension to not be available).

Also, we should explicitly list these dependencies in the README.

Thoughts? @stripe/api-libraries

**Comment by kyleconroy** [comment]
Seems like a reasonable addition. @bkrausz may have more state on if this is the best

**Comment by bkrausz** [comment]
The only mbstring method we use is `mb_detect_encoding` to decide if we should `utf8_encode`. We can instead add a `function_exists('mb_detect_encoding')` to that call (preferably with class var caching since I think the perf of `function_exists` sucks and it'd be called a lot). May still be worth a warning about poor UTF8 handling, but if they don't have `mbstring` installed I think they already have UTF8 problems.

Also putting in `init` would only work for those not using Composer, so we'd likely want somewhere else (like in ApiRequestor, which always gets called on request).

I think we used to do a check for Curl, but Composer enforces its existence, and we now provide pluggable request handlers, so you could opt to swap in your own non-Curl requestor if you're so inclined.

**Comment by matthewarkin** [comment]
does the ext-mbstring dependency in the composer.json (https://github.com/stripe/stripe-php/blob/master/composer.json#L21) not handle the composer case?

**Comment by GrahamCampbell** [comment]
Composer already deals with this. Nothing else is needed IMO.

**Comment by olivierbellone** [comment]
Tentatively fixed by #251.
